recipe/meta.yaml

```diff
@@ -0,0 +1,161 @@
+{% set name = "torchaudio" %}
+{% set version = "2.1.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pytorch/audio/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 82c2745a73172b495327ec36c6970ad5ad9d5d2ac44feeaea2617152f9393bf7
+  patches:
+    - 0001-set-include-path.patch
+
+build:
+  number: 0
+  skip: True  # [py>311]
+  # temporarily skip windows until ffmpeg is available
+  script:
+    - export USE_FFMPEG=ON          # [not win]
+    - set USE_FFMPEG=ON             # [win]
+    - export BUILD_SOX=OFF          # [not win]
+    - set BUILD_SOX=OFF             # [win]
+    - export FFMPEG_ROOT=${PREFIX}  # [not win]
+    - set FFMPEG_ROOT=${PREFIX}     # [win]
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+    - install_name_tool -add_rpath @loader_path/ $SP_DIR/torchaudio/lib/_torchaudio_ffmpeg.so  # [osx]
```

Comment:
isn't conda-build supposed to do all this? Seems strange why it isn't... I wonder if there's a better way to do this...

Reply:
These are the python extensions. It's a little bit special in this case. The code first manually loads `libtorchaudio`, then imports `torchaudio.lib._torchaudio`. See https://github.com/pytorch/audio/blob/v2.1.2/torchaudio/_extension/__init__.py#L44-L47. Because `libtorchaudio` is loaded first, there will be runtime error that says it can't be found when `torchaudio.lib._torchaudio` is imported. Same for the other libraries, I bet pytorch does a similar dance since torchaudio copied their dance.

Reply:
the tests pass without this (smoke and unit tests, not sure whether unit tests use pytorch though) at least on osx: https://github.com/AnacondaRecipes/torchaudio-feedstock/pull/2

other platforms are failing for unrelated reasons (ffmpeg staging channel doesn't contain linux builds for some reason; syntax in the test section is unix for windows platform).

As JC pointed out offline, nevertheless we should understand what's going on and why setting the rpath might be unnecessary and not cause problems.

Reply:
I don't fully understand how not setting those rpaths would be **unnecessary**. If the binary was built such that it points to those `.so` files, I think we should add the `RPATH` for it. Whether there's a more elegant way to do this, I don't know.

Reply:
if I understand correctly, it looks like the python code is loading the libraries rather than the loader (i.e. the system runtime)

Reply:
Exactly that. See https://github.com/AnacondaRecipes/torchaudio-feedstock/pull/1#discussion_r1679941293.